src/Controllers/MovieController.cs

```diff
@@ -247,5 +251,57 @@ public IActionResult Details(int? id)
                 return NotFound();
             }
         }
+        
+        [HttpGet("SendMovieMail")]
+        public IActionResult SendMovieMail(int? id)
+        {
+            try 
+            {
+                if (id == null)
+                    throw new NullReferenceException();
+                var movie = UnitOfWork.MovieRepository.Get(id.Value);
+                if (movie == null)
+                    throw new NullReferenceException();
+                MovieViewModel movieVM = new MovieViewModel
+                {
+                    Id = movie.Id,
+                    Title = movie.Title,
+                    ReleaseDate = movie.ReleaseDate,
+                    Genre = movie.Genre,
+                    Price = movie.Price,
+                    Rating = movie.Rating,
+                };
+                return View(movieVM);
+            }
+            catch(NullReferenceException)
+            {
+                return NotFound();
+            }
+        }
+
+        [HttpPost("SendMovieMail")]
+        public IActionResult SendMovieMail([FromForm] string emailAddress, int? id)
+        {
+            try 
+            {
+                if (id == null)
+                    throw new NullReferenceException();
+                var movie = UnitOfWork.MovieRepository.Get(id.Value);
+                if (movie == null)
+                    throw new NullReferenceException();
+                StringBuilder body = new StringBuilder();
```

Comment:
Try appending the whole body as one string

Reply:
I made a commit with the change